src/object/srv_obj.c

```diff
@@ -2355,6 +2429,10 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 		dss_get_module_info()->dmi_xs_id, orw->orw_epoch,
 		orw->orw_map_ver, ioc.ioc_map_ver, DP_DTI(&orw->orw_dti));
 
+	if (obj_rpc_is_fetch(rpc) && !(orw->orw_flags & ORF_EC_RECOV) &&
+	    (orw->orw_epoch != 0 && orw->orw_epoch != DAOS_EPOCH_MAX))
+		ioc.ioc_fetch_snap = 1;
```

Comment:
is this the safe way to tell if this is snapshot epoch? Probably you can compare the orw_epoch with the most recent snapshot epoch above?

Reply:
OK, probably can be refine as that - only for degraded fetch case, fetch snap list from IV (if snap list unavialable) and compare with the orw_epoch. I can refine it on a following patch. thx